pkg/agent/agent.go

```diff
@@ -0,0 +1,159 @@
+// Package agent implements functions for injecting faults in a target
+package agent
+
+import (
+	"context"
+	"fmt"
+	"syscall"
+	"time"
+
+	"github.com/grafana/xk6-disruptor/pkg/agent/protocol"
+	"github.com/grafana/xk6-disruptor/pkg/agent/protocol/grpc"
+	"github.com/grafana/xk6-disruptor/pkg/agent/protocol/http"
+	"github.com/grafana/xk6-disruptor/pkg/runtime"
+)
+
+// Config maintains the configuration for the execution of the agent
+type Config struct {
+	Profiler *runtime.ProfilerConfig
+}
+
+// Agent maintains the state required for executing an agent command
+type Agent struct {
+	env    runtime.Environment
+	config *Config
+}
+
+// BuildAgent builds a instance of an agent
+func BuildAgent(env runtime.Environment, config *Config) *Agent {
+	return &Agent{
+		env:    env,
+		config: config,
+	}
+}
+
+// do executes a command in the Agent
+func (r *Agent) do(ctx context.Context, action func(context.Context) error) error {
```

Comment:
I think this `action` parameter really wants to become an interface, as said above.

Reply:
Not sure I understand, can you please elaborate which interface, with wich methods?